source/docs/stats.md

```diff
@@ -130,8 +130,21 @@ across the codebase.
 `const StatName` member variables. Most names should be established during
 process initializion or in response to xDS updates.
 
-`StatNameSet` provides some associative lookups at runtime, using two maps: a
-static map and a dynamic map.
+`StatNameSet` provides some associative lookups at runtime. The associations
+should be created before the set is used for requests, via
+`StatNameSet::rememberBuiltin`. This is useful in scenarios where stat-names are
+derived from data in a request, but there are limited set of known tokens, such
+as SSL ciphers or Redis commands.
+
+### Dynamic stat tokens
```

Comment:
Can you potentially add a section in stats.md about the encoding we are using in each scenario? It's complex enough at this point that I think a general treatment of what we are doing and how we are doing it would be super useful. In particular it would be useful to cover how dynamic/builtin stats can be joined, interact, etc.